Library/Homebrew/cask/lib/hbc/cli/reinstall.rb

```diff
@@ -7,6 +7,24 @@ def self.install_casks(cask_tokens, force, skip_cask_deps, require_sha)
           begin
             cask = Hbc.load(cask_token)
 
+            installer = Installer.new(cask,
+                                      force:          force,
+                                      skip_cask_deps: skip_cask_deps,
+                                      require_sha:    require_sha)
+            installer.print_caveats
+
+            # Download
+            begin
+              installer.satisfy_dependencies
+              installer.verify_has_sha if @require_sha && !@force
+              installer.download
+              installer.verify
+            rescue StandardError => e
+              installer.purge_versioned_files
```

Comment:
Don't rescue at all, because not rescuing at all is the same as rescuing and re-raising.